### PR TITLE
Alternative shouldReceive syntax

### PR DESCRIPTION
**Pull request by davedevelopment**
Not sure what people think about this, it just gets away from using strings for method names...

``` php
$mock->shouldReceive()->foo("bar")->andReturn(123);
```

### CODE REVIEW

File: library/Mockery/Mock.php [hunk 1]
```diff
@@ -178,10 +179,10 @@ public function mockery_init(\Mockery\Container $container = null, $partialObjec
      * @param string $methodName,... one or many methods that are expected to be called in this mock
      * @return \Mockery\Expectation
      */
-    public function shouldReceive($methodName)
+    public function shouldReceive($methodName = null)
```
Comment (GrahamCampbell) [inline library/Mockery/Mock.php]: php doc needs updating to string|null

File: library/Mockery/ExpectationBuilder.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{
+    private $mock;
+
+    public function __construct(MockInterface $mock)
+    {
+        $this->mock = $mock;
+    }
+
+    /**
+     * @return Mockery\Expectation
```
Comment (GrahamCampbell) [inline library/Mockery/ExpectationBuilder.php]: missing param annotations

**Comment by davedevelopment** [comment]
``` php
<?php

require "vendor/autoload.php";

$mock = Mockery::mock();
                                                 
$mock->shouldReceive()->foo(123)->andReturn(456);
$mock->shouldNotReceive()->bar(567);

var_dump($mock->foo(123));

$mock->shouldHaveReceived()->foo(123);
$mock->shouldNotHaveReceived()->bar(456);

Mockery::close();
```

**Comment by davedevelopment** [comment]
Riffing on the same thing, I could leave the old school methods alone and do 

``` php
<?php

require "vendor/autoload.php";

$mock = Mockery::mock();

// $mock->shouldReceive("foo")->with(123)->once()->andReturn(456);
$mock->expect()->foo(123)->andReturn(456);

// $mock->shouldReceive("bar")->with(567)->andReturn(789);
$mock->allow()->bar(567)->andReturn(789);
```

Might collide with some userland code. Not sure what equivalent I'd want to use for the spies.

**Comment by usox** [comment]
+1
A bit late but I would really prefer having real methods instead of dealing with method names a string-parameters.

**Pull request merged**